doc/tutorial/calcium_stdp_curve.rst

```diff
@@ -0,0 +1,185 @@
+synapse dynamics described in Graupner and Brunel [1]_.
+
+The synapse is modeled as synaptic efficacy variable, :math:`\rho`, which is a function of the
+calcium concentration, :math:`c(t)`. There are two stable states at :math:`\rho=0` (DOWN) and
+:math:`\rho=1` (UP), while :math:`\rho=\rho^\ast=0.5` represents a third unstable state between the
+two stable states.  The calcium concentration dynamics are represented by a simplified model which
+uses a linear sum of individual calcium transients elicited by trains of pre- and postsynaptic
+action potentials:
+
+.. math::
+
+   \begin{align*}
+   c^\prime (t) &= - \frac{1}{\tau_{Ca}}c 
+                   + C_{pre} \sum_i \delta(t-t_i-D)
+                   + C_{post} \sum_j \delta(t-t_j), \\
+   \rho^\prime(t) &= - \frac{1}{\tau}\left [
+                     \rho (1 - \rho) (\rho^\ast - \rho)
+                     -\gamma_p (1-\rho) H\left(c - \theta_p \right)
+                     + \gamma_d \rho H\left(c - \theta_d \right) \right ]
+                     + N, \\
+   N &= \frac{\sigma}{\sqrt{\tau}} \sqrt{H\left( c - \theta_p \right)
+        + H\left( c - \theta_d \right)} W.
+   \end{align*}
+
+Here, the sums over :math:`i` and :math:`j` represent the contributions from all pre- and
+postsynaptic spikes, respectively, with :math:`C_{pre}` and :math:`C_{pre}` denoting the jumps in
+concentration after a spike. The jump after the presynaptic spike is delayed by :math:`D`.  The
+Calcium decay time is assumed to be much faster than the synaptic time scale,
+:math:`\tau_{Ca} \ll \tau`. The subscripts :math:`p` and :math:`d` represent potentiation (increase
+in synaptic efficacy) and depression (decrease in synaptic efficacy), respectively, with
+:math:`\gamma` and :math:`\theta` being the corresponding rates and thresholds. :math:`H(x)` is the
+right-continuous heaviside step function (:math:`H(0)=1`).
+
+This mechanism is stochastic, :math:`W` represents a white noise process, and therefore our
+simulation needs to
+
+- use a stochastic synapse mechanism,
+- accumulate statistics over a large enough ensemble of initial states.
+
+
+Implementation of a Stochastic Mechanism
+----------------------------------------
+
+Implementing a stochastic mechanism which is given by a stochastic differential equation (SDE) as
+above is straightforward to implement in Arbor's NMODL dialect. Let's examine the mechanism code 
+in the `Arbor repository <https://github.com/arbor-sim/arbor/mechanisms/stochastic/calcium_based_synapse.mod>`_.
+
+The main difference compared to a deterministic (ODE) description is the additional `WHITE_NOISE`
+block,
+
+.. code:: none
+
+   WHITE_NOISE {
+       W
+   }
+
+which declares the white noise process :math:`W`, and the specification pf the `stochastic` solver
+method,
+
+.. code:: none
+
+   BREAKPOINT {
+       SOLVE state METHOD stochastic
+   }
+
+This is sufficient to inform Arbor about the stochasticity of the mechanism.
```

Comment:
can we add a link to an explanation of what kind of solver `stochastic` implements? https://docs.arbor-sim.org/en/latest/dev/numerics.html

Reply:
Yes you are right, that should be mentioned. I added links to the longer explanations of how we treat SDEs but also added an entry to the `dev/numerics.rst` - this document was actually commented out, so I re-added it to the TOC.